### DISCUSSION TITLE
Component rerenders when input value is cleared ?

### DISCUSSION THREAD

**Discussion by LhonRafaat** [react-hook-form/react-hook-form#8489] [Q&A]
Hello there,  
I'm using useFormContext and custom Input component which is basically just a designed input  to do a basic edit form control inside a modal. 
I use setValue to set initial values for my form fields inside a useEffect which is dependant on a prop change : 

```
useEffect(() => {
    setValue('cost', dailyExpense.cost);
    setValue('description', dailyExpense.description);
    setValue('name', dailyExpense.name);
    setValue('spentBy', dailyExpense.spentBy);
  }, [dailyExpense]);
```

and this is my input component : 

```
const Input = React.forwardRef<HTMLInputElement, Props>(({ variant, searchClassName, ...rest }, ref) => {
  if (variant === 'search') {
    return (
      <div className={`relative ${searchClassName}`}>
        <input
          style={{ paddingLeft: isRtl ? '0' : '24px', paddingRight: isRtl ? '24px' : '0' }}
          className="rounded-full placeholder-current text-gray-600  xl:px-4 py-2 xl:min-w-[350px] pr-10  bg-white border border-inputBorder px-4  outline-none  focus:border-purple-600 transition h-12 w-full "
          {...rest}
        />
        <SearchIcon
          className="absolute top-4 xl:top-4 w-4 h-4 "
          style={{
            right: isRtl ? undefined : '16px',
            left: isRtl ? '16px' : undefined,
          }}
        />
      </div>
    );
  }
  return (
    <input
      ref={ref}
      className={`bg-white border border-inputBorder px-4 rounded-lg outline-none  focus:border-purple-600 transition h-12 ${classes}`}
      {...rest}
      onWheel={(e) => e.currentTarget.blur()}
    />
  );
});
```
```
 <Input
          {...register('name', { required: true })}
          defaultValue={dailyExpense.name}
          variant="default"
          placeholder="Eg. restructuring the garage"
          id="email"
        />
```

so , my issue is when I edit of of the input values (try to clear it) it rerenders the component which makes the setValue inside the useEffect run again and block the input from clearing as the value resets again. I tried to just use defaultValue on the inputs but it just causes some updating problems , the values dont set correctly when switching between rows (the rows I edit). I couldnt find an alternative for the initialValues, since im using the useFormContext () , any advices of what is done wrong here ? 

thanks ,
![rerender](https://user-images.githubusercontent.com/61140151/173364711-4da5b3cf-c2e6-465f-9a0e-d17cafd5c9ea.png)

**Comment by bluebill1049** [comment]
if you want to update lots of fields, maybe `reset` is better?

```tsx
reset({
  ...getValues(),
  updatedValue: 'a',
  updatedValue1: 'b',
})
```

**Reply by LhonRafaat** [reply]
![Screenshot from 2022-06-14 09-26-45](https://user-images.githubusercontent.com/61140151/173508111-8d481875-7f39-4458-a7ec-714c8864a6ad.png)


that does not set the initialValues for me , 



```
useEffect(() => {
    () =>
      reset({
        ...getValues(),
        cost: dailyExpense.cost,
        description: dailyExpense.description,
        name: dailyExpense.name,
        spentBy: dailyExpense.spentBy,
      });
  }, [dailyExpense]);
```


also if I dont use the reset with an function call it gets into an infinite loop , 

any ideas?
